### ISSUE TITLE
How does cabal select a package from multiple global repositories?

### ISSUE THREAD

**Issue by TerrorJack** [haskell/cabal#6139]
When multiple repositories (e.g. both head.hackage and regular hackage) are present in `~/.cabal/config`, a single package might come from different repos, how does `cabal` decide from which repo is the package picked up? The [documentation](https://github.com/haskell/cabal/blob/master/Cabal/doc/installing-packages.rst#repository-specification) only says:

> If you want, you can configure multiple repositories, and cabal will combine them and be able to download packages from any of them.

The intuition would be selecting from the first repo which contains the package, but it's actually not the case, e.g. when running `cabal new-install dlist` or `cabal new-build` for a local package depending on `dlist`, the `dlist` package from head.hackage is *not* picked up, even when head.hackage comes before regular hackage in the global config.

**Comment by gbaz** [comment]
Note that there is a ghc proposal pending review that addresses the lack of control over this: https://github.com/ghc-proposals/ghc-proposals/pull/115

**Comment by 23Skidoo** [comment]
IIRC it depends on the order the repos are declared in the config file, I think the last one wins.

**Comment by phadej** [comment]
Yes, last one wins. I'd already implemented a `priority` field, but that effort is blocked on e.g. https://github.com/haskell/cabal/issues/6101 (I really don't want to touch any of `ReadP` parsers at this point of time).

**Comment by fgaz** [comment]
i think the recently released `active-repositories` fixes this
